src/data/ar-io-data-source.ts

```diff
@@ -399,15 +408,31 @@ export class ArIODataSource
     requestStartTime,
     peer,
     ttfb,
+    expectedHash,
   }: {
     response: AxiosResponse;
     requestAttributes: RequestAttributes;
     requestStartTime: number;
     peer: string;
     ttfb: number;
+    expectedHash?: string;
   }): ContiguousData {
     const stream = response.data;
 
+    // Check if peer's digest matches our expected hash
+    if (expectedHash !== undefined) {
+      const peerDigest = response.headers[headerNames.digest.toLowerCase()];
```

Comment:
it says in axios docs https://axios-http.com/docs/res_schema

```
  // All header names are lower cased and can be accessed using the bracket notation.
```

but this isn't like computationally heavy thing to do, but this could be done without the toLowerCase strictly speaking.